spec/system/admin/reports_spec.rb

```diff
@@ -752,7 +752,7 @@ def xero_invoice_summary_row(description, amount, tax_type, opts = {})
 
     def xero_invoice_li_row(line_item, opts = {})
       tax_type = line_item.has_tax? ? 'GST on Income' : 'GST Free Income'
-      xero_invoice_row line_item.product.sku, line_item.product_and_full_name,
+      xero_invoice_row line_item.variant.sku, line_item.product_and_full_name,
                        line_item.price.to_s, line_item.quantity.to_s, tax_type, opts
```

Comment:
Another confusing thing: Products and Variants both have an SKU. Before the Product SKU was stored on the master variant though. Now it's explicitly on the Product model.

Still, not sure why this update was needed. The product should still have an SKU, just stored differently.

Reply:
It's because the test setup here was previously using `product.master` as the _variant_ that was being assigned to the line items being tested. I updated that in a previous commit, I think. 

Basically I found there were a lot of old tests where the test setup was essentially wrong as they were using the master variant as a kind of shortcut when setting up an order object with line items, but essentially those tests should have been updated during the big Spree upgrade (~2.5 years ago?) because master variants do not get associated to line items under any circumstances, and haven't done for a long time.

Does that make sense?

Reply:
I think it does, Thanks !